external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -114,7 +121,10 @@ object SchemaConverters {
       case ByteType | ShortType | IntegerType => builder.intType()
       case LongType => builder.longType()
       case DateType => builder.longType()
-      case TimestampType => builder.longType()
+      case TimestampType =>
+        // To be consistent with the previous behavior of writing Timestamp type with Avro 1.7,
```

Comment:
the previous behavior is: we can't write out timestamp data, isn't it?

Reply:
also we should follow parquet and have a config `spark.sql.avro.outputTimestampType` to control it.

Reply:
Previously we write timestamp as `Long` and divide the value by 1000(millisecond precision).
Maybe I need to revise the comment.
+1 on the new config.

Reply:
For now I think writing out timestamp micros should be good